pkg/resource/deploy/plan_executor.go

```diff
@@ -126,7 +126,7 @@ func (pe *planExecutor) Execute(callerCtx context.Context, opts Options, preview
 	//     should bail.
 	//  3. The stepExecCancel cancel context gets canceled. This means some error occurred in the step executor
 	//     and we need to bail. This can also happen if the user hits Ctrl-C.
-	canceled, err := func() (bool, error) {
+	canceled, res := func() (bool, *result.Result) {
```

Comment:
isn't 'caceled' redundant with 'res'?  perhaps consdier updating the comment above.